### PR TITLE
Overload defineConfig to accept a RollupOptionsFunction parameter

### PR DESCRIPTION
**Pull request by Septh**
This PR contains:

- [ ] bugfix
- [x] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

List any relevant issue numbers:

### Description

This PR provides a `defineConfigFunction` auxiliary function, akin to `defineConfig` but for configs that exports a function:

```js
// Adapted from last example at https://rollupjs.org/guide/en/#configuration-files
export default defineConfigFunction(commandLineArgs => {
  const inputBase = commandLineArgs.input || 'main.js';
  delete commandLineArgs.input;
  return {
    input: 'src/entries/' + inputBase,
    output: {...}
  }
})
```

### CODE REVIEW

**Comment by Septh** [comment]
I'm not really fond of the naming...

**Review by lukastaegert — COMMENTED** [review]
Great idea

> I'm not really fond of the naming...

neither am I. Why not just overload `defineConfig` a third time to also accept a `RollupConfigFunction` as parameter?

**Comment by Septh** [comment]
> Why not just overload `defineConfig` a third time to also accept a `RollupConfigFunction` as parameter?
Why making it simple when we can make it complicated? 😃 

You're absolutely right, I'm updating the PR ASAP. 🙏

**Comment by Septh** [comment]
Also renamed the type from `RollupConfigFunction` to `RollupOptionsFunction`, for consistency.

File: test/load-config-file/index.js [hunk 1]
```diff
@@ -41,6 +41,15 @@ describe('loadConfigFile', () => {
 		assert.deepStrictEqual(JSON.parse(JSON.stringify(options)), defaultConfigs);
 	});
 
+	it('loads an ESM config file exporting a config with defineConfigFn()', async () => {
```
Comment (lukastaegert) [inline test/load-config-file/index.js:44]: Minor nit: The two test descriptions no longer fit.

**Review by lukastaegert — APPROVED** [review]
Looks great to me, consider if you want to improve the test descriptions, will merge this in the next days.

**Review by lukastaegert — APPROVED** [review]
Awesome 🎉

**Pull request merged**